modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx

```diff
@@ -0,0 +1,168 @@
+import NumberFormat from 'react-number-format';
+
+import PropTypes from "prop-types";
+
+import MultipleChoice from "./MultipleChoice";
+import Question from "./Question";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+
+// Component that renders a multiple choice question, with optional number input.
+// Selected answers are placed in a series of <input type="hidden"> tags for
+// submission.
+//
+// Optional props:
+//  max: Integer denoting maximum number of options that may be selected
+//  min: Integer denoting minimum number of options that may be selected
+//  name: String containing the question to ask
+//  defaults: Array of objects, each with an "id" representing internal ID
+//            and a "value" denoting what will be displayed
+//  userInput: Either "input", "textbox", or undefined denoting the type of
+//             user input. Currently, only "input" is supported
+//  maxValue: The maximum allowed input value
+//  minValue: The minimum allowed input value
+//  type: One of "integer" or "float" (default: "float")
+//  errorText: String to display when the input is not valid (default: "invalid input")
+//
+// Sample usage:
+// <NumberQuestion
+//    name="Please enter the patient's age"
+//    defaults={[
+//      {"id": "<18", "label": "<18"}
+//    ]}
+//    max={1}
+//    minValue={18}
+//    type="integer"
+//    errorText="Please enter an age above 18, or select the <18 option"
+//    />
+function NumberQuestion(props) {
+  let {defaults, max, min, name, userInput, minValue, maxValue, type, errorText, ...rest} = props;
+  const [error, setError] = useState(false);
+
+  // Callback function for our min/max
+  let checkNumber = (text) => {
+    let value = 0;
+
+    if (typeof(text) === "undefined") {
+      // The custom input has been unset
+      setError(false);
+      return
+    }
+
+    if (type === "integer") {
+      // Test that it is an integer
+      if (!/^[-+]?\d*$/.test(text)) {
+        setError(true);
+        return;
+      }
+
+      value = parseInt(text);
+    } else if (type === "float") {
+      value = Number(text);
+
+      // Reject whitespace and non-numbers
+      if (/^\s*$/.test(text) || isNaN(value)) {
+        setError(true);
+        return;
+      }
+    }
+
+    // Test that it is within our min/max (if they are defined)
+    if ((typeof minValue !== 'undefined' && value < minValue) ||
+      (typeof maxValue !== 'undefined' && value > maxValue)) {
+      setError(true);
+      return;
+    }
+    setError(false);
+  }
+
+  return (
+    <Question
+      text={name}
+      >
+      {error && <Typography color='error'>{errorText}</Typography>}
+      <MultipleChoice
+        max={max}
+        min={min}
+        defaults={defaults}
+        input={userInput==="input"}
+        textbox={userInput==="textbox"}
+        onChange={checkNumber}
+        additionalInputProps={{
+          min: minValue,
+          max: maxValue,
+          allowNegative: minValue < 0,
```

Comment:
When not specifying a `minValue` at all, negatives are not allowed.

```suggestion
          allowNegative: (minValue === undefined || minValue < 0),
```